classes/CustomText.php

```diff
@@ -39,11 +39,116 @@ class CustomText extends ObjectModel
 		'table' => 'info',
 		'primary' => 'id_info',
 		'multilang' => true,
+		'multilang_shop' => true,
 		'fields' => array(
-			'id_shop' =>			array('type' => self::TYPE_NOTHING, 'validate' => 'isUnsignedId'),
+			'id_info' =>	array('type' => self::TYPE_NOTHING, 'validate' => 'isUnsignedId'),
+			'id_shop' =>	array('type' => self::TYPE_NOTHING, 'validate' => 'isUnsignedId'),
 			// Lang fields
-			'text' =>				array('type' => self::TYPE_HTML, 'lang' => true, 'validate' => 'isCleanHtml', 'required' => true),
+			'text' =>		array('type' => self::TYPE_HTML, 'lang' => true, 'validate' => 'isCleanHtml', 'required' => true),
 		)
 	);
 
+	/**
+	 * Adding new custom text.
+	 * @param bool $autoDate
+	 * @param bool $nullValues
+	 * @return bool
+	 * @throws PrestaShopDatabaseException
+	 * @throws PrestaShopException
+	 */
+	public function add($autoDate = true, $nullValues = false)
+	{
+		$this->id = $this->id_info;
+
+		if (!$result = Db::getInstance()->insert($this->def['table'], $this->getFields(), $nullValues)) {
+			return false;
+		}
+
+		// Database insertion for multilingual fields related to the object
+		if (!empty($this->def['multilang'])) {
+			$fields = $this->getFieldsLang();
+			if ($fields && is_array($fields)) {
+				$asso = Shop::getAssoTable($this->def['table'].'_lang');
+				foreach ($fields as $field) {
+					foreach (array_keys($field) as $key) {
+						if (!Validate::isTableOrIdentifier($key)) {
+							throw new PrestaShopException('key '.$key.' is not table or identifier');
+						}
+					}
+					$field[$this->def['primary']] = (int)$this->id;
+
+					if (false !== $asso && 'fk_shop' === $asso['type']) {
+						$field['id_shop'] = (int)$this->id_shop;
+						$result &= Db::getInstance()->insert($this->def['table'] . '_lang', $field);
+					} else {
+						$result &= Db::getInstance()->insert($this->def['table'].'_lang', $field);
+					}
+				}
+			}
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Save the custom text changes.
+	 * @param bool $null_values
+	 * @param bool $auto_date
+	 * @return bool
+	 */
+	public function save($null_values = false, $auto_date = true)
+	{
+		$this->id = $this->id_info;
```

Comment:
$this->id doesnt exists anymore, you're removed it :)

but the important point here is that you don't use nor $this->id or $this->id_info, so why do you affect it?